src/kernel/src/main.rs

```diff
@@ -535,3 +473,89 @@ impl Default for ExecutionEngine {
         ExecutionEngine::Llvm
     }
 }
+
+#[derive(Debug, Error)]
+enum DiscordPresenceError {
+    #[error("Failed to create Discord IPC")]
+    FailedToCreateIpc(#[source] Box<dyn Error>),
+
+    #[error("Failed to update Discord presence")]
+    FailedToUpdatePresence(#[source] Box<dyn Error>),
+}
+
+#[derive(Debug, Error)]
+enum KernelError {
+    #[error("Failed to open .kernel-debug")]
+    FailedToOpenDebug(#[source] std::io::Error),
+
+    #[error("Failed to parse .kernel-debug")]
+    FailedToParseDebug(#[from] serde_yaml::Error),
+
+    #[error("Failed to parse arguments")]
+    FailedToParseArgs(#[from] clap::Error),
+
+    #[error("Failed to open param.sfo")]
+    FailedToOpenParam(#[source] std::io::Error),
+
+    #[error("Failed to read param.sfo")]
+    FailedToReadParam(#[from] param::ReadError),
+
+    #[error("{0} has invalid title identifier.")]
+    InvalidTitleId(PathBuf),
+
+    #[error("Filesystem initialization failed")]
+    FilesystemInitFailed(#[from] FsError),
+
+    #[error("Memory manager initialization failed")]
+    MemoryManagerInitFailed(#[from] MemoryManagerError),
+
+    #[error("Failed to initialize TtyManager")]
+    TtyInitFailed(#[from] TtyError),
+
+    #[error("Virtual process initialization failed")]
+    VProcInitFailed(#[from] VProcError),
+
+    #[error("Runtime linker initialization failed")]
+    RuntimeLinkerInitFailed(#[source] Box<dyn Error>),
+
+    #[error("Failed to load libkernl")]
+    FailedToLoadLibkernel(#[source] Box<dyn Error>),
+
+    #[error("Failed to load libSceLibcInternal")]
+    FailedToLoadLibSceLibcInternal(#[source] Box<dyn Error>),
+
+    #[error("Failed to create main thread")]
+    FailedToCreateMainThread(#[from] SpawnError),
+
+    #[error("Failed to join with main thread")]
+    FailedToJoinMainThread(#[source] std::io::Error),
+}
+
+// We have to use this for a custom implementation of the Termination trait, because
+// we need to log the error using our own error! macro instead of std::fmt::Debug::fmt,
+// which is what the default implementation of Termination uses for Result<T: Termination, E: Debug>.
```

Comment:
Why using comments instead of a documentation?

Reply:
I'm not really used to writing it, so I didn't realized that was even something I should do :D.
Is it acceptable now?.